core/src/block.rs

```diff
@@ -616,7 +618,7 @@ impl CandidateBlock {
             .map(|accepted_tx| {
                 accepted_tx.and_then(|tx| {
                     transaction_validator
```

Comment:
this pattern and the fact that you had a problem finding a bug makes me think we should have `block_validator` instead of `transaction_validator` which would always clone wsv internally. I would appreciate if you can make this PR include this change

Reply:
Let's make it in a separate pull request. It doesn't seem easy.